main.py

```diff
@@ -292,6 +293,30 @@ def disable_switch(self, request: Request) -> JSONResponse:
         self.notify_topology_update()
         self.notify_switch_links_status(switch, "link disabled")
         return JSONResponse("Operation successful", status_code=201)
+    
+    @rest('v3/switches/{dpid}', methods=['DELETE'])
+    def delete_switch(self, request: Request) -> JSONResponse:
+        """Delete switch"""
+        dpid = request.path_params["dpid"]
+        with self._switch_lock:
+            try:
+                switch: Switch = self.controller.switches[dpid]
+            except KeyError:
+                raise HTTPException(404, detail="Switch not found.")
+            if switch.status != EntityStatus.DISABLED:
+                raise HTTPException(409, detail=f"The switch is not disable.")
+            links = self.links.copy()
+            for link_id, link in links.items():
+                if link.endpoint_a.switch.dpid == dpid or link.endpoint_b.switch.dpid == dpid:
+                    raise HTTPException(409, detail=f"This switch have link {link_id}")
+            switch = self.controller.switches.pop(dpid)
```

Comment:
In the issue https://github.com/kytos-ng/topology/issues/164 there was an initial question what to do with flows. I believe the most conservative approach would be to get `GET v2/stored_flows?dpid=dpid` and make sure that dpid doesn't have any installed flows, if it does, then recommend that network operators disable the services that installed these flows, `mef_eline` and `telemetry_int` once EVCs get deleted they'll remove them, similarly `of_lldp` will do it when a switch gets disabled, but `coloring` doesn't yet, so `coloring` needs to do that as a pre-requisite too, just so we have all NApps behaving coherently in terms of installing/removing flows to avoid surprises and new bug reports here. See what I mean?

Reply:
Now checking for any flows. There should be none. Timeout argument is missing.

Reply:
Right. Also, related to that:

- Let's take the opportunity to also document in a new issue why we're choosing this more conservative approach, just so if folks in the future ask for trying to auto delete a `Switch` topology resource we can explain the challenges and why it's going to be this way for now. In summary, we're following the current NApp responsibilities where each NApp should be responsible for cleaning up their resources and we're also relying heavily on the fact that a topology resource needs to be disabled upfront. 
- Replace `requests` with `httpx` (see comment below)
- It might be worth also including the request into the `self._switch_lock[dpid]` scope, since essentially the entire logic is that it's supposed to be atomic by switch, it's a minor detail, but it can safe a few requests from being sent in a worst case high concurrent scenario with the same dpid. Also, arguably since the request is a bit more expensive than the other runtime checks regarding the switch state, it might be worth moving it to only happen after those checks on line 330

Reply:
Replaced `requests` with `httpx` and using a dictionary for `_switch_lock`. It is missing the documentation.

Reply:
Issue created [#182](https://github.com/kytos-ng/topology/issues/182)

Reply:
Nice. `flows = self.get_flows_by_switch(dpid)` can get moved a few lines down below. We know it's not be such expensive request, but still, there are other checks that you're doing that are lighter so those should be first, and then the request later, because if those checks were to fail then you just made a request in vain, see what I mean? Minor stuff, but that'd be great.

Reply:
Moved as last verification.